core/application/commands.go

```diff
@@ -37,6 +37,29 @@ func (a *app) CreateMicroVM(ctx context.Context, mvm *models.MicroVM) (*models.M
 		return nil, fmt.Errorf("an error occurred when attempting to validate microvm spec: %w", validErr)
 	}
 
+	if mvm.Spec.RootVolume.Source.VirtioFS != nil {
+		return nil, fmt.Errorf("VirtioFS is not available yet for RootVolume")
+	}
+	if len(mvm.Spec.AdditionalVolumes) > 0 {
+		virtioFSCount := 0 // Counter for VirtioFS volumes
+		
+		for _, volume := range mvm.Spec.AdditionalVolumes {
+			if volume.Source.Container != nil && volume.Source.VirtioFS != nil {
+				return nil, fmt.Errorf("Cannot have both ContainerSource and VirtioFS on the same volume")
+			}
+			
+			// Check if this volume has VirtioFS
+			if volume.Source.VirtioFS != nil {
+				virtioFSCount++
+			}
+		}
+	
+		// Ensure only one volume contains VirtioFS
+		if virtioFSCount > 1 {
```

Comment:
We could add this check to the "additional volumes" as a custom validator as well.

Reply:
Can you check if the most recent code update resolves this?